assets/hip-206/solidity/IHederaTokenService.sol

```diff
@@ -49,6 +49,168 @@ interface IHederaTokenService {
+    }
+
+    /// A fee to assess during a transfer that changes ownership of an NFT. Defines the fraction of
+    /// the fungible value exchanged for an NFT that the ledger should collect as a royalty. ("Fungible
+    /// value" includes both ℏ and units of fungible HTS tokens.) When the NFT sender does not receive
+    /// any fungible value, the ledger will assess the fallback fee, if present, to the new NFT owner.
+    /// Royalty fees can only be added to tokens of type type NON_FUNGIBLE_UNIQUE.
+    struct RoyaltyFee {
+        // A fraction's numerator of fungible value exchanged for an NFT to collect as royalty
+        uint32 numerator;
+
+        // A fraction's denominator of fungible value exchanged for an NFT to collect as royalty
+        uint32 denominator;
+
+        // If present, the fixed fee to assess to the NFT receiver when no fungible value is exchanged
+        // with the sender
+        FixedFee fixedFee;
+
+        // The ID of the account to receive the custom fee, expressed as a solidity address
+        address feeCollector;
+    }
+
+    /// A Key can be a public key from either the Ed25519 or ECDSA(secp256k1) signature schemes, where
+    /// in the ECDSA(secp256k1) case we require the 33-byte compressed form of the public key. We call
+    /// these public keys <b>primitive keys</b>.
+    /// A Key can also be the ID of a smart contract instance, which is then authorized to perform any 
+    /// precompiled contract action that requires this key to sign. 
+    /// Note that when a Key is a smart contract ID, it <i>doesn't</i> mean the contract with that ID 
+    /// will actually create a cryptographic signature. It only means that when the contract calls a 
+    /// precompiled contract, the resulting "child transaction" will be authorized to perform any action
+    /// controlled by the Key.
+    /// Exactly one of the possible values should be populated in order for the Key to be valid.
+    struct Key {
+        // smart contract instance that is authorized as if it had signed with a key
+        address contractId;
+
+        // Ed25519 public key bytes
+        bytes ed25519;
+
+        // Compressed ECDSA(secp256k1) public key bytes
+        bytes ECDSA_secp256k1;
+
+        // A smart contract that, if the recipient of the active message frame, should be treated
+        // as having signed. (Note this does not mean the <i>code being executed in the frame</i> 
+        // will belong to the given contract, since it could be running another contract's code via 
+        // <tt>delegatecall</tt>. So setting this key is a more permissive version of setting the
+        // contractID key, which also requires the code in the active message frame belong to the
+        // the contract with the given id.)
+        address delegatableContractId;
+    }
+
+    /// A convenience struct used to group different types of Keys we want to set to a Token during
+    /// token creation.
+    struct TokenKeys{
```

Comment:
Rather than a struct with all of the possible keys what about a struct with a key type (one of admin, kyc, ...) and then the Key.  The API would accept an array of this key struct.

And then, what if we made the type actually a bit field, so you could set multiple keys with one entry.  Odds are most contracts will set all to one, or one to each.  For the all to one scenario we save a lot of gas.